embassy-stm32/src/rcc/h7/mod.rs

```diff
@@ -508,6 +510,172 @@ impl<'d> Rcc<'d> {
         }
     }
 }
+pub enum McoClock {
+    Disabled,
+    Bypassed,
+    Divided(u8),
+}
+
+impl McoClock {
+    fn into_raw(&self) -> u8 {
+        match self {
+            McoClock::Disabled => 0,
+            McoClock::Bypassed => 1,
+            McoClock::Divided(divisor) => {
+                if *divisor > 15 {
+                    panic!("Mco divisor must be less than 15. Refer to the reference manual for more information.")
+                }
+                *divisor
+            }
+        }
+    }
+}
+
+#[derive(Copy, Clone)]
+pub enum Mco1Source {
+    Hsi,
+    Lse,
+    Hse,
+    Pll1Q,
+    Hsi48,
+}
+
+impl Default for Mco1Source {
+    fn default() -> Self {
+        Self::Hsi
+    }
+}
+
+pub trait McoSource: Sized + Default + Copy {
+    fn into_raw(&self) -> u8;
+}
+
+impl McoSource for Mco1Source {
+    fn into_raw(&self) -> u8 {
+        match self {
+            Mco1Source::Hsi => 0,
+            Mco1Source::Lse => 1,
+            Mco1Source::Hse => 2,
+            Mco1Source::Pll1Q => 3,
+            Mco1Source::Hsi48 => 4,
+        }
+    }
+}
+
+#[derive(Copy, Clone)]
+pub enum Mco2Source {
+    SysClk,
+    Pll2Q,
+    Hse,
+    Pll1Q,
+    Csi,
+    Lsi,
+}
+
+impl Default for Mco2Source {
+    fn default() -> Self {
+        Self::SysClk
+    }
+}
+
+impl McoSource for Mco2Source {
+    fn into_raw(&self) -> u8 {
+        match self {
+            Mco2Source::SysClk => 0,
```

Comment:
I also wonder if there should be an `enum` within the `rcc_h7.yaml` register?

Reply:
it'd be nice, but I wouldn't require doing it *now*, we can always add it later.